### DISCUSSION TITLE
Generating a CLI file

### DISCUSSION THREAD

**Discussion by eflehder1** [leap71/PicoGK#49] [Coding for (Computational) Engineers]
Hi @leap71 , I am struggling with the transfer of a large geometry file for the purpose of getting a quote. STL files become very large when making a large porous structure of small pores. I have been recommended to generate a CLI file instead and thought I would ask for your advice on this? I understand that I could use Magics or other software for this but wondered if there is a way to do it within PicoGK or an open-source alternative?

**Comment by LinKayser** [comment]
Hi Eric — good news, you can literally just use Voxels.SaveToCliFile(), you can check out the example: https://github.com/leap71/PicoGK/blob/main/Examples/Ex_PrepForPrintSTL.cs

You need to ask the print provider for the layer height they want to use and provide that as a parameter.

Best,
Lin

**Reply by eflehder1** [reply]
Hi Lin, thats perfect thank you very much!

Best,
Eric
